src/bin/gosub-parser.rs

```diff
@@ -41,14 +39,22 @@ fn main() -> Result<()> {
         chars.detect_encoding();
     }
 
-    let document = DocumentBuilder::new_document();
-    let parse_errors = Html5Parser::parse_document(&mut chars, Document::clone(&document), None)?;
+    // Create a new document that will be filled in by the parser
+    let handle = DocumentBuilder::new_document();
+    let parse_errors = Html5Parser::parse_document(&mut chars, Document::clone(&handle), None)?;
 
-    println!("Generated tree: \n\n {document}");
+    let mut binding = Document::clone(&handle);
+    let mut handle_mut = binding.get_mut();
```

Comment:
You can remove one clone here, however you'd need to make handle mutable.

```suggestion
    let mut handle_mut = handle.get_mut();
```
if you don't want to do it, maybe rename the variable.